### PR TITLE
A bug template to make introducing bugs a more streamlined process

### PR DESCRIPTION
**Pull request by SCdF**
I want to introduce a bug reporting template to see if we can help speed up the triage / reproduction phase of bug reporting, and help everyone write consistent, helpful and thorough bug reports.

Interesting in your thoughts on wording / process etc.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -1 +1,26 @@
...
+ - _The problem_
+
+**What should happen**:
+ - _Detail what you **expect** to happen when you complete the last step_
+
+**What actually happens**:
+ - _Describe what happens (the bug!)_
+
+**Environment**:
+_Provide environmental information. Try to be as thorough as possible. What user, what OS, browser, server, app version, config version etc._
```
Comment (garethbowen) [inline .github/ISSUE_TEMPLATE.md]: I would give more structure here, eg:

```
- Instance: (alpha.dev.medicmobile.org, etc)
- Browser: (Firefox, Chrome, incognito mode, etc)
- Client platform: (Windows, MacOS, Linux)
- App: (webapp, admin, sentinel, api, couch2pg, medic-conf, etc)
- Version: (2.15.0, 3.0.0, etc)
```

Shamelessly stolen from [PouchDBs template](https://raw.githubusercontent.com/pouchdb/pouchdb/master/ISSUE_TEMPLATE.md).

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -1 +1,26 @@
...
+ - _Write individual steps here_
+ - _That can be used to reproduce_
+ - _The problem_
+
+**What should happen**:
+ - _Detail what you **expect** to happen when you complete the last step_
+
+**What actually happens**:
+ - _Describe what happens (the bug!)_
+
```
Comment (garethbowen) [inline .github/ISSUE_TEMPLATE.md]: Have a section for logs - people always forget to include the logs. Also list the places the logs are found (browser console, gardener, nginx, couch, etc).

File: .github/ISSUE_TEMPLATE.md [hunk 3]
```diff
@@ -1 +1,26 @@
...
+**Steps to reproduce**:
+ - _Write individual steps here_
+ - _That can be used to reproduce_
+ - _The problem_
+
+**What should happen**:
+ - _Detail what you **expect** to happen when you complete the last step_
+
+**What actually happens**:
+ - _Describe what happens (the bug!)_
```
Comment (garethbowen) [inline .github/ISSUE_TEMPLATE.md]: If possible include a link to a doc/page that exhibits this bug - this would save me so much time!

Comment (garethbowen) [reply .github/ISSUE_TEMPLATE.md]: Also, screenshots!

**Comment by abbyad** [comment]
I fully support an improved template for bug/issue reporting. We should get folks outside of the dev team who submit bugs to also review this new version as well. There may be things that we are asking for which wouldn't be obvious how to get for a non-dev, and we could either give instructions as notes , eg `<!-- how to get ____ -->`, or link to documentation.

File: .github/ISSUE_TEMPLATE.md (hunk 2 above)
Comment (abbyad) [reply .github/ISSUE_TEMPLATE.md]: I would make sure to document _how_ to get logs, either in the issue template or link to relevant documentation. Also, not all those reporting bugs would have access to logs... so we should request the URL and whoever is investigating can access the logs.

**Pull request merged**